brownie/_cli/analyze.py

```diff
@@ -0,0 +1,268 @@
+passed explicitly with the respective command line option.
+
+Alternatively, you have to provide a username/password combination. It
+is recommended to pass them through the environment variables as
+"MYTHX_ETH_ADDRESS" and "MYTHX_PASSWORD".
+
+You can also choose to not authenticate and submit your analyses as a free
+trial user. No registration required! To see your past analyses, get access
+to deeper vulnerability detection, and a neat dashboard, register at
+https://mythx.io/. Any questions? Hit up dominik.muhs@consensys.net or contact
+us on the website!
+"""
+
+
+SEVERITY_COLOURS = {Severity.LOW: "green", Severity.MEDIUM: "yellow", Severity.HIGH: "red"}
+DASHBOARD_BASE_URL = "https://dashboard.mythx.io/#/console/analyses/"
+TRIAL_PRINTED = False
+BYTECODE_ADDRESS_PATCH = re.compile(r"__\w{38}")
+DEPLOYED_ADDRESS_PATCH = re.compile(r"__\$\w{34}\$__")
+
+
+def construct_source_dict_from_artifact(artifact):
+    return {
+        artifact.get("sourcePath"): {
+            "source": artifact.get("source"),
+            # "ast": artifact.get("ast"),  # NOTE: Reenable once container issue fixed
+        }
+    }
+
+
+def construct_request_from_artifact(artifact):
+    global BYTECODE_ADDRESS_PATCH
+
+    bytecode = artifact.get("bytecode")
+    deployed_bytecode = artifact.get("deployedBytecode")
+    source_map = artifact.get("sourceMap")
+    deployed_source_map = artifact.get("deployedSourceMap")
+
+    bytecode = re.sub(BYTECODE_ADDRESS_PATCH, "0" * 40, bytecode)
+    deployed_bytecode = re.sub(DEPLOYED_ADDRESS_PATCH, "0" * 40, deployed_bytecode)
+
+    source_list = artifact.get("allSourcePaths")
+    return {
+        "contract_name": artifact.get("contractName"),
+        "bytecode": bytecode if bytecode else None,
+        "deployed_bytecode": deployed_bytecode if deployed_bytecode else None,
+        "source_map": source_map if source_map else None,
+        "deployed_source_map": deployed_source_map if deployed_source_map else None,
+        "sources": construct_source_dict_from_artifact(artifact),
+        "source_list": source_list if source_list else None,
+        "main_source": artifact.get("sourcePath"),
+        "solc_version": artifact["compiler"]["version"].replace("Version:", "").strip(),
+        "analysis_mode": "full" if ARGV["full"] else "quick",
+    }
+
+
+def get_mythx_client():
+    # if both eth address and username are None,
+    if ARGV["access-token"]:
+        authenticated = True
+        auth_args = {"access_token": ARGV["access-token"]}
+    elif environ.get("MYTHX_ACCESS_TOKEN"):
+        authenticated = True
+        auth_args = {"access_token": environ.get("MYTHX_ACCESS_TOKEN")}
+    elif environ.get("MYTHX_ETH_ADDRESS") and environ.get("MYTHX_PASSWORD"):
+        authenticated = True
+        auth_args = {
+            "eth_address": environ.get("MYTHX_ETH_ADDRESS"),
+            "password": environ.get("MYTHX_PASSWORD"),
+        }
+    elif ARGV["eth-address"] and ARGV["password"]:
```

Comment:
I would swap the positions of lines 108-110 and 102-107 so we have consistent behavior with cli arguments over-riding environment variables.